src/Plugin/Validators/EmptyCell.php

```diff
@@ -78,12 +104,137 @@ public function validateRow($row_values) {
+   *   - If the case string returned by the validator implied validation passed.
+   *   - If the case string returned by the validator is not recognized.
+   */
+  public static function processListWithDescribedTable(array $validation_results, array $tokens = [], array $metadata = []) {
+
+    // Define our table header.
+    $table_header = ['Line Number', 'Column(s) with empty value'];
+    $table['rows'] = [];
+
+    $default_tokens = array_column(self::$mapping, 'default-msg', 'token');
+    // Combine our provided and our default token arrays. Because array_merge
+    // will overwrite values in the first array with values from the second
+    // array for the same keys, we provide our default tokens first.
+    $combined_tokens = array_merge($default_tokens, $tokens);
+
+    foreach ($validation_results as $line_no => $validation_status) {
+      // Check the format of the validation_status parameter.
+      ImportValidationHelper::checkValidationStatusArray($validation_status, 'EmptyCell', $line_no);
+
+      if ($validation_status['case'] == self::$mapping['case-empty-value']['dev-case']) {
+        $table['message'] = $combined_tokens['case-empty-value'];
+        // Convert indices in failedItems to column headers.
+        $failed_indices = $validation_status['failedItems']['empty_indices'];
+        // For each index with an empty value, grab the column name from our
+        // $headers property and add to an array of header names.
+        $empty_headers = [];
+        foreach ($failed_indices as $index) {
+          array_push($empty_headers, $metadata['column_headers'][$index]);
+        }
+        // Implode the empty headers array into a string and then add it as a
+        // row to our table.
+        $columns_string = implode(", ", $empty_headers);
+        array_push($table['rows'], [
+          $line_no,
+          $columns_string,
+        ]);
+      }
+      elseif ($validation_status['case'] == self::$mapping['case-valid']['dev-case']) {
+        throw new \Exception("The case string returned by the EmptyCell validator at line #$line_no implies validation passed, but valid is set to FALSE.");
+      }
+      else {
+        throw new \Exception("The case string returned by the EmptyCell validator at line #$line_no is not recognized as a potential case.");
+      }
+    }
+
+    // Now replace any tokens that are in our message or items.
+    // We use the Tripal Token Parser service to ensure that more complicated
+    // tokens are supported.
+    // NOTE: Dependency injection is NOT used since this is a static method.
+    $service_TripalTokensParser = \Drupal::service('tripal.token_parser');
+    $replaced_message = $service_TripalTokensParser->replaceTokens($table['message'], $combined_tokens);
+
+    // Build the render array for our table.
+    $render_array = [
+      '#theme' => 'item_list',
+      '#type' => 'ul',
+      '#attributes' => [
+        'class' => [
+          'tcp-empty-cell-failures',
```

Comment:
Just noticed this one- EmptyCell is now in TripalCultivate 🙈 
```suggestion
          'tc-empty-cell-failures',
```